frontend/src/theme/ApiDemoPanel/Curl/index.js

```diff
@@ -0,0 +1,300 @@
+
+const languageSet = [
+  {
+    tabName: 'cURL',
+    highlight: 'bash',
+    language: 'curl',
+    variant: 'curl',
+    options: {
+      longFormat: false,
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+  {
+    tabName: 'Node',
+    highlight: 'javascript',
+    language: 'nodejs',
+    variant: 'axios',
+    options: {
+      ES6_enabled: true,
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+  {
+    tabName: 'Go',
+    highlight: 'go',
+    language: 'go',
+    variant: 'native',
+    options: {
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+  {
+    tabName: 'Python',
+    highlight: 'python',
+    language: 'python',
+    variant: 'requests',
+    options: {
+      followRedirect: true,
+      trimRequestBody: true,
+    },
+  },
+]
+const languageTheme = {
+  plain: {
+    color: 'var(--ifm-code-color)',
+  },
+  styles: [
+    {
+      types: ['inserted', 'attr-name'],
+      style: {
+        color: 'var(--openapi-code-green)',
+      },
+    },
+    {
+      types: ['string', 'url'],
+      style: {
+        color: 'var(--openapi-code-green)',
+      },
+    },
+    {
+      types: ['builtin', 'char', 'constant', 'function'],
+      style: {
+        color: 'var(--openapi-code-blue)',
+      },
+    },
+    {
+      types: ['punctuation', 'operator'],
+      style: {
+        color: 'var(--openapi-code-dim)',
+      },
+    },
+    {
+      types: ['class-name'],
+      style: {
+        color: 'var(--openapi-code-orange)',
+      },
+    },
+    {
+      types: ['tag', 'arrow', 'keyword'],
+      style: {
+        color: 'var(--openapi-code-purple)',
+      },
+    },
+    {
+      types: ['boolean'],
+      style: {
+        color: 'var(--openapi-code-red)',
+      },
+    },
+  ],
+}
+
+function getBaseUrl() {
+  /*
+  This is a special case for production.
+
+  We want to be able to build the front end with no value set for
+  `BASE_URL` so that staging, prod and self hosting users
+  can all use the same docker image.
+
+  When deployed to staging, we want the frontend on
+  https://staging.shields.io/ to generate badges with the base
+  https://staging.shields.io/
+  (and we want similar behaviour for users hosting their own instance)
+
+  When we promote to production we want https://shields.io/ and
+  https://www.shields.io/ to both generate badges with the base
+  https://img.shields.io/
+
+  For local dev, we can deal with setting the api and front-end
+  being on different ports using the BASE_URL env var
+  */
+  const { protocol, hostname, port } = window.location
+  if (['shields.io', 'www.shields.io'].includes(hostname)) {
+    return 'https://img.shields.io'
+  }
+  if (!port) {
+    return `${protocol}//${hostname}`
+  }
+  return `${protocol}//${hostname}:${port}`
+}
+
+function getServer() {
+  return {
+    url: getBaseUrl(),
+    variables: {},
+  }
+}
+
+function Curl({ postman, codeSamples }) {
+  // TODO: match theme for vscode.
```

Comment:
Is this a real todo for us or a copy pasta from somewhere else?

Reply:
Nope it is from the upstream
https://github.com/cloud-annotations/docusaurus-openapi/blob/0beb11b248e34e2b88f171406fc4fb04a390f065/packages/docusaurus-theme-openapi/src/theme/ApiDemoPanel/Curl/index.tsx#L126

In https://github.com/badges/shields/pull/9014#discussion_r1142363731
I've touched on the topic of why I've tried to keep the changes I've made minimal in the section starting "There are 2 competing objectives here..."
Essentially: if a new version of the upstream package is released, I want it to be reasonably easy to diff our customised version against the package version to see if there are any changes we need to make (this is something I'd like to make some CI tooling for, but.. later).